src/BugsnagUnity.m

```diff
@@ -754,22 +767,28 @@ void bugsnag_retrieveLastRunInfo(const void *lastRuninfo, void (*callback)(const
 
 char * bugsnag_retrieveDeviceData(const void *deviceData, void (*callback)(const void *instance, const char *key, const char *value)) {
     BugsnagDeviceWithState *device = [Bugsnag.client generateDeviceWithState:BSGGetSystemInfo()];
-    NSDictionary *deviceDictionary = [NSDictionary dictionaryWithObjectsAndKeys:
-        device.freeDisk, @"freeDisk",
-        device.freeMemory, @"freeMemory",
-        device.id, @"id",
-        device.jailbroken ? @"true" : @"false", @"jailbroken",
-        device.locale, @"locale",
-        device.manufacturer, @"manufacturer",
-        device.model, @"model",
-        device.modelNumber, @"modelNumber",
-        device.runtimeVersions[@"osBuild"], @"osBuild",
-        device.osName, @"osName",
-        device.osVersion,@ "osVersion",
-        nil];
+    NSMutableDictionary *deviceDictionary = [[NSMutableDictionary alloc] init];
+
+    if (device.freeDisk != nil) [deviceDictionary setObject:device.freeDisk forKey:@"freeDisk"];
+    if (device.freeMemory != nil) [deviceDictionary setObject:device.freeMemory forKey:@"freeMemory"];
+    if (device.id != nil) [deviceDictionary setObject:device.id forKey:@"id"];
+    if (device.jailbroken) {
+        [deviceDictionary setObject:@"true" forKey:@"jailbroken"];
+    } else {
+        [deviceDictionary setObject:@"false" forKey:@"jailbroken"];
+    }
```

Comment:
Small nit (not mandatory): the ternary you used for `app.isLaunching` seems a cleaner way to represent this.